.gitignore

```diff
@@ -7,3 +7,14 @@ coverage
 stunnel.conf
 stunnel.pid
 *.out
+package-lock.json
+
+# IntelliJ IDEs
+.idea
+
+# VisualStudioCode IDEs
+.vscode/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
```

Comment:
Nit: adding all IDE parts is not possible. Thus, it's normally ideal to let users ignore these globally on their local machine.

Reply:
It's fairly common practice to add these in on OSS repositories, though I'll get rid of the `!.vscode*` lines as they're not required